R/score.R

```diff
@@ -0,0 +1,57 @@
+# score.R
+
+#' Produce a gH object: a hash of genes and their associated scores
+#' (likelihood that observed mutations exceed background mutations at FDR of 1%)
+#'
+
+score <- function(gX, gH, silent = FALSE, noLog = FALSE) {
+    
+    check parameters with .checkArgs()
+    
+    open gX file
+    
+    # for each gene, check if it already in the gH hash and if yes, do not calculate its score (in case have a repeat of a gene in gX, would just use first calculated score)
+    
+    for gene in gX {

```

Comment:
I would iterate through the entire data frame (it's an rds object, so you get the whole thing at once anyway), and collect the information as you go through it. So perhaps build your hash from the genes you find in ```unique(gX$sym)```, and then go through the frame row by row.